### PR TITLE
For expert review

### PR DESCRIPTION
**Pull request by DemienBlogan**

### CODE REVIEW

File: StringList.c [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+	// node[0] - value of the node
+	// node[1] - pointer to the next node
+
+	// Creating node:
+	char** newNode = (char**)malloc(sizeof(char*) * 2);
+
+	// Node value:
+	size_t strLength = strlen(str);
+	newNode[0] = (char*)malloc(sizeof(char) * (strLength + 1));
+	strcpy(newNode[0], str);
```
Comment (Stepbura) [inline StringList.c:61]: The potential null pointer is passed into 'strcpy' function. Inspect input arguments.

File: StringList.c [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+Return type: index of found string (-1 if there's no match or if the list is empty)
+Description: Find string in the list of the strings (using Linear Search)
+*/
+int FindNode(const char* const* const* list, const char* str)
+{
+	// *list   - first node
+	// node[0] - value of the node
+	// node[1] - pointer to the next node
+
+	if (*list == NULL || list == NULL)
```
Comment (Stepbura) [inline StringList.c:119]: Please verify if 'str' is not null as well

File: StringList.c [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+Return type: true - the string was deleted, false - the string wasn't found
+Description: Try to delete string from the list of string.
+*/
+bool DeleteNode(char*** list, size_t* size, const char* str)
+{
+	// *list   - first node
+	// node[0] - value of the node
+	// node[1] - pointer to the next node
+
+	if (*list == NULL || list == NULL)
```
Comment (Stepbura) [inline StringList.c:152]: Please verify if 'str' is not null as well

File: StringList.c [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+
+			if (currectNode == *list)
+			{
+				char** temp = *list;
+				*list = (*list)[1];
+				free(temp);
+			}
+			else
+			{
+				previousNode[1] = currectNode[1];
```
Comment (Stepbura) [inline StringList.c:172]: Dereferencing of the null pointer 'previousNode' might take place.

File: StringList.c [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+
+	while (currectNode != NULL)
+	{
+		if (!strcmp(currectNode[0], oldStr))
+		{
+			free(currectNode[0]);
+
+			size_t strLength = strlen(newStr);
+			currectNode[0] = (char*)malloc(sizeof(char) * (strLength + 1));
+			strcpy(currectNode[0], newStr);
```
Comment (Stepbura) [inline StringList.c:251]: The potential null pointer is passed into 'strcpy' function. Inspect input arguments.

**Review by Stepbura — CHANGES_REQUESTED** [review]
Please take a look my comments.

File: StringList.c (hunk 1 above)
Comment (DemienBlogan) [reply StringList.c:64]: Solved. Thanks.

File: StringList.c (hunk 2 above)
Comment (DemienBlogan) [reply StringList.c]: Solved. Thanks.

File: StringList.c (hunk 3 above)
Comment (DemienBlogan) [reply StringList.c]: Solved. Thanks.

File: StringList.c (hunk 4 above)
Comment (DemienBlogan) [reply StringList.c]: Solved. Thanks.

File: StringList.c (hunk 5 above)
Comment (DemienBlogan) [reply StringList.c:256]: Solved. Thanks.

**Review by Stepbura — APPROVED** [review]

**Pull request merged**